### PR TITLE
fix: make background loops work with get_data_set.

### LINKED ISSUE #395: Fix #394 for #389 does not work with get_data_set
A minimal example:

```
import time
import numpy as np
import qcodes
from qcodes.tests.data_mocks import DataSet2D
from qcodes.tests.instrument_mocks import DummyInstrument

dataset = DataSet2D()
A = DummyInstrument(server_name='test')
sweepvalues=A.dac2[0:100:.05]

#%%
#%Loop in background
print(qcodes.get_bg())

loop=qcodes.Loop(sweepvalues, progress_interval=.2, delay=.001).each(A.dac1)
data=loop.get_data_set() # data_manager=True)
print('created dataset...')
#data=loop.run(background=True)
data=loop.run(background=True, data_manager=True)
#data=loop.run()
print('sync %s' % data.sync())
print('fraction %.3f' % data.fraction_complete())
data.complete() # should wait on data to be ready

for ii in range(20):    
    data.sync()
    print('sync %s' % data.sync())
    print('fraction %.3f' % data.fraction_complete())
    time.sleep(0.2)
```

### PR DESCRIPTION
**Pull request by alexcjohnson**
Fixes #395 

Changes proposed in this pull request:
- Look for existing background loops during `get_data_set` rather than during `run`

@peendebak @giulioungaretti

### CODE REVIEW

**Comment by giulioungaretti** [comment]
Will it fix? Merging!

**Comment by alexcjohnson** [comment]
> test were passing locally ?

blurg no. sorry, will sort it out.

**Comment by alexcjohnson** [comment]
OK, fixed.

File: qcodes/loops.py [hunk 1]
```diff
@@ -723,6 +735,18 @@ def get_data_set(self, data_manager=USE_MP, *args, **kwargs):
...
+    def wait_for_bg(self, quiet):
+        prev_loop = get_bg()
+        if prev_loop:
+            if not quiet:
+                print('Waiting for the previous background Loop to finish...',
+                      flush=True)
+            prev_loop.join()
+
+        if prev_loop and not quiet:
+            print('...done. Starting ' + (data_set.location or 'new loop'),
```
Comment (giulioungaretti) [inline qcodes/loops.py]: will this actually print or raise an exception ?

**Pull request closed without merging**